### PR TITLE
Tweaks to XCM for Benchmarking

### PR DESCRIPTION
**Pull request by shawntabrizi**
This PR extracts minor refactorings used in https://github.com/paritytech/polkadot/pull/3940.

The only logical change happening here is the introduction of `is_subscribed` API to the XCM versioning handler.

* Moved `TestSubscriptionService` and `TestAssetTrap` into a `test_utils` file so it can be reused.
* Moved logic after `xcm::execute` to `xcm::post_execute` so that it can be triggered manually.
* Added `is_subscribed` to XCM Version Subscribtion API

### CODE REVIEW

**Review by KiChjang — APPROVED** [review]

**Review by apopiak — COMMENTED** [review]
Just some quick drive-by comments :sweat_smile:

File: xcm/xcm-executor/src/lib.rs [hunk 1]
```diff
@@ -228,6 +209,29 @@ impl<Config: config::Config> XcmExecutor<Config> {
 		result
 	}
 
+	pub fn post_execute(mut self, xcm_weight: Weight) -> Outcome {
```
[the remark this answers is not in the document]
Comment (shawntabrizi) [reply xcm/xcm-executor/src/lib.rs:212]: because i want to use it in benchmarking

[the remark this answers is not in the document]
Comment (shawntabrizi) [reply xcm/xcm-executor/src/lib.rs:212]:
```suggestion
	/// Execute any final operations after having executed the XCM message.
	/// This includes refunding surplus weight, trapping extra holding funds, and returning any errors during execution.
	pub fn post_execute(mut self, xcm_weight: Weight) -> Outcome {
```

**Review by apopiak — APPROVED** [review]
LGTM

**Comment by shawntabrizi** [comment]
bot merge

**Comment by shawntabrizi** [comment]
bot merge

**Review by Jaysonmaldx — APPROVED** [review]

**Comment by Jaysonmaldx** [comment]
Thank u

**Pull request merged**